### DISCUSSION TITLE
How to define Non-argument dependencies for Graph-backed assets?

### DISCUSSION THREAD

**Discussion by PascalStehling** [dagster-io/dagster#10029] [Q&A] [answered]
Hi,

Is there a way to define Non-argument dependencies for Graph-backed assets defined with AssetsDefinition.from_graph? I could not find an Option in you documentation for this.

Keep on the good work,
Best Pascal

**Comment by dpeng817** [comment, marked as answer]
Hey Pascal - We don't have a mechanism for providing asset dependencies without an argument to a graph, but you can use arguments to define the dependency structure without actually passing data. To do this, you would use nothing inputs on the ops within your graph, and then pass the input argument from your graph to those ops:
```python
from dagster import op, graph, Nothing, In, AssetsDefinition

@op(ins={"noop": In(Nothing)})
def the_op():
    ... # do stuff

@graph
def the_graph(my_asset_dependency_no_data):
    the_op(my_asset_dependency_no_data)

graph_backed_asset = AssetsDefinition.from_graph(the_graph) # assumes the input name is the asset key
```
